MudSelect: Inherit Style from its MudComponentBase

pupper9 opened the pull request:
<!--
MAKE SURE TO READ THE CONTRIBUTING GUIDE BEFORE CREATING A PR
https://github.com/MudBlazor/MudBlazor/blob/dev/CONTRIBUTING.md

Testing sections can be removed for documentation changes
-->

<!-- Provide a general summary of your changes in the Title above -->
<!-- Keep the title short and descriptive, as it will be used as a commit message -->


## Description
<!-- Describe your changes in detail any why -->
<!-- Note any issues that are resolved by this PR -->
<!-- e.g. resolves #1337 or fixes #9310 -->

MudSelect does not apply the styles provided to it's Style property because it doesn't get passed to its InputContent from the MudComponentBase.

This commit addresses that.

## How Has This Been Tested?
<!-- All PR's should implement unit tests if possible -->
<!-- Please describe how you tested your changes. -->
<!-- Have you created new tests or updated existing ones? -->
<!-- e.g. unit | visually | none -->

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

<!-- If you made any visual changes, provide screenshots of before/after, it its moving parts, please provide high quality gif, wemb or mp4 -->

## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [ ] I've added relevant tests.

pupper9 committed:
MudSelect: Inherit Style from its MudComponentBase

ScarletKuro commented:
@henon not sure if this is correct, since now the Style is applied in both places `MudInputControl` and `MudInput`.

henon commented:
It definitely bears the risk of unexpected results. In other components we solved similar problems by adding another style parameter. Adding an optional `InputStyle` parameter would be the safe way to go.

ScarletKuro commented:
Yeah, I also think `InputStyle` is way to go. @pupper9 can you modify your code?

pupper9 committed:
Added optional InputStyle to MudInput and MudSelect so that MudInput can have it's own style independent of MudSelect's Style.

pupper9 commented:
Ok @henon and @ScarletKuro I added the optional InputStyle parameter to MudSelect and MudInput and reverted the original change.

I wasn't sure if I should have added the InputStyle to MudBaseInput or MudInput, I didn't go with MudBaseInput because I think this was a specific issue for MudSelect.

henon commented:
continued as PR https://github.com/MudBlazor/MudBlazor/pull/7542
